---
output: html_document
editor_options: 
  chunk_output_type: console
---
...
title: "rMATS analysis"
author: "Shay Iyer"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
...

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(maser)
library(rtracklayer)
library(gridExtra) 
```

```{r maser path definition}
ips.s1s2.path <- "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/001_ips-s1s2/003_ips-s1s2_output/002_ips-s1s2_rmats-out"
```

```{r maser objects definition}
#using JCEC for statistical power
ips.s1s2 <- maser(path = ips.s1s2.path, cond_labels = c("iPS", "S1S2-iEC"), ftype = "JCEC")
```

# PreProcessing for mapping script, 10/22/22

```{r}
SE <- summary(ips.s1s2, type = "SE")[] %>% as.data.frame() %>% mutate("EventType" = rep("SE")) %>% filter(Chr == "chr19") %>% filter(FDR < 0.05)
MXE <- summary(ips.s1s2, type = "MXE")[] %>% as.data.frame() %>% mutate("EventType" = rep("MXE")) %>% filter(Chr == "chr19") %>% filter(FDR < 0.05)
A5SS <- summary(ips.s1s2, type = "A5SS")[] %>% as.data.frame() %>% mutate("EventType" = rep("A5SS"))%>% filter(Chr == "chr19") %>% filter(FDR < 0.05)
A3SS <- summary(ips.s1s2, type = "A3SS")[] %>% as.data.frame() %>% mutate("EventType" = rep("A3SS")) %>% filter(Chr == "chr19") %>% filter(FDR < 0.05)
```

```{r}
rm(ips.s1s2, ips.s1s2.path)
```

```{r}
SE %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/seToy.csv")
MXE %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/mxeToy.csv")
A5SS %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/a5ssToy.csv")
A3SS %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/a3ssToy.csv")
```

```{r not using retained intron for 10/23/22}
RI <- summary(ips.s1s2, type = "RI")[] %>% as.data.frame() %>% mutate("EventType" = rep("RI")) 
RI %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/riToy.csv")
```


```{r}
sdypath <- "/Volumes/sheynkman/projects/shay_thesis/data/EC-LR/long-read-EC-data/03_chr19_gtfs/chr19_ENCFF370SDY.gtf"
cgkpath <- "/Volumes/sheynkman/projects/shay_thesis/data/EC-LR/long-read-EC-data/03_chr19_gtfs/chr19_ENCFF422CGK.gtf"
uqopath <- "/Volumes/sheynkman/projects/shay_thesis/data/EC-LR/long-read-EC-data/03_chr19_gtfs/chr19_ENCFF544UQO.gtf"
aykpath <- "/Volumes/sheynkman/projects/shay_thesis/data/EC-LR/long-read-EC-data/03_chr19_gtfs/chr19_ENCFF610AYK.gtf"
dwvpath <- "/Volumes/sheynkman/projects/shay_thesis/data/EC-LR/long-read-EC-data/03_chr19_gtfs/chr19_ENCFF798DWV.gtf"
```

```{r}
sdygtf <- rtracklayer::import(sdypath)
cgkgtf <- rtracklayer::import(cgkpath)
uqogtf <- rtracklayer::import(uqopath)
aykgtf <- rtracklayer::import(aykpath)
dwvgtf <- rtracklayer::import(dwvpath)
```

```{r}
#cleaning
rm(sdypath, cgkpath, uqopath, aykpath, dwvpath)
```

```{r}
sdy = sdygtf %>% as.data.frame() 
cgk = cgkgtf %>% as.data.frame()
uqo = uqogtf %>% as.data.frame() 
ayk = aykgtf %>% as.data.frame()
dwv = dwvgtf %>% as.data.frame()
```

```{r}
rm(aykgtf, cgkgtf, dwvgtf, sdygtf, uqogtf)
```

```{r}
sdy = sdy %>% as.data.frame() %>% select(seqnames, start, end, width, strand, type, gene_id, gene_name, exon_id, exon_number, transcript_id, transcript_name) %>% mutate("source" = "sdy")
cgk = cgk %>% as.data.frame() %>% select(seqnames, start, end, width, strand, type, gene_id, gene_name, exon_id, exon_number, transcript_id, transcript_name) %>% mutate("source" = "cgk")
uqo = uqo %>% as.data.frame() %>% select(seqnames, start, end, width, strand, type, gene_id, gene_name, exon_id, exon_number, transcript_id, transcript_name) %>% mutate("source" = "uqo")
ayk = ayk %>% as.data.frame() %>% select(seqnames, start, end, width, strand, type, gene_id, gene_name, exon_id, exon_number, transcript_id, transcript_name) %>% mutate("source" = "ayk")
dwv = dwv %>% as.data.frame() %>% select(seqnames, start, end, width, strand, type, gene_id, gene_name, exon_id, exon_number, transcript_id, transcript_name) %>% mutate("source" = "dwv")
```

```{r}
sdy %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/sdy.csv")
cgk %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/cgk.csv")
uqo %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/uqo.csv")
ayk %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/ayk.csv")
dwv %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/dwv.csv")
```

```{r}
aykshort = ayk %>% head(20)
aykshort %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/aykshort.csv")
```

```{r finding a filter candidate gene}
ayk_genes <- ayk$gene_name
SE_genes <- SE$geneSymbol
MXE_genes <- MXE$geneSymbol
A3SS_genes <- A3SS$geneSymbol
A5SS_genes <- A5SS$geneSymbol

shared <- intersect(SE_genes, MXE_genes)


shared <- intersect(ayk_genes, SE_genes) %>% unique()

shared_ayk <- ayk %>% filter(gene_name %in% shared)
shared_SE <- SE %>% filter(geneSymbol %in% shared)  

sharedayk_counts <- shared_ayk %>% group_by(gene_name) %>% summarize(n = n()) %>% arrange(desc(n))

sharedSE_counts <- shared_SE %>% group_by(geneSymbol) %>% summarize(n=n()) %>% arrange(desc(n))

```

```{r}
SE %>% filter(geneSymbol == "TCF3") %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/seTCF3.csv")
```


```{r}
sdy %>% filter(gene_name == "TCF3") %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/sdyTCF3.csv")
cgk %>% filter(gene_name == "TCF3") %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/cgkTCF3.csv")
uqo %>% filter(gene_name == "TCF3") %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/uqoTCF3.csv")
ayk %>% filter(gene_name == "TCF3") %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/aykTCF3.csv")
dwv %>% filter(gene_name == "TCF3") %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/dwvTCF3.csv")
```

