---
output: html_document
editor_options: 
  chunk_output_type: console
---
...
title: "rMATS analysis"
author: "Shay Iyer"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
...

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(maser)
library(rtracklayer)
library(gridExtra) 
```

> Goal of analysis: find statistically significant splicing events associated with endothelial cell differentiation

```{r maser path definition}
ips.s1s2.path <- "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/001_ips-s1s2/003_ips-s1s2_output/002_ips-s1s2_rmats-out"
ips.s1etv2.path <- "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/002_ips-s1etv2/003_ips-s1etv2_output/002_ips-s1etv2_rmats-out"
s1etv2.ecfc.path <- "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/003_s1etv2-ecfc/003_s1etv2-ecfc_output/002_s1etv2-ecfc_rmats-out"
```

```{r maser objects definition}
#using JCEC for statistical power
ips.s1s2 <- maser(path = ips.s1s2.path, cond_labels = c("iPS", "S1S2-iEC"), ftype = "JCEC")
```


```{r maser objects definition}
ips.s1etv2 <- maser(path = ips.s1etv2.path, cond_labels = c("iPS", "S1ETV2"), ftype = "JCEC")
s1etv2.ecfc <- maser(path = s1etv2.ecfc.path, cond_labels = c("S1ETV2", "ECFC"), ftype = "JCEC")
```

```{r}
ips.s1s2.table <- summary(ips.s1s2) %>% as.data.frame()
```

```{r}
ips.s1etv2.table <- summary(ips.s1etv2) %>% as.data.frame()
s1etv2.ecfc.table <- summary(s1etv2.ecfc) %>% as.data.frame()

n_distinct(ips.s1s2.table$GeneID)
n_distinct(ips.s1etv2.table$GeneID)
n_distinct(s1etv2.ecfc.table$GeneID)
```

```{r maser object filtering}
#ips.s1s2
#filtering for low coverage reads (!)
f.ips.s1s2 <- filterByCoverage(ips.s1s2, avg_reads=5) 
#selecting top events by FDR <= 0.05 and deltapsi >= 0.1
top.ips.s1s2 <- topEvents(f.ips.s1s2, fdr = 0.05, deltaPSI = 0.1)

#ips.s1etv2
f.ips.s1etv2 <- filterByCoverage(ips.s1etv2, avg_reads=5)
top.ips.s1etv2 <- topEvents(f.ips.s1etv2, fdr = 0.05, deltaPSI = 0.1)

#s1etv2.ecfc
f.s1etv2.ecfc <- filterByCoverage(s1etv2.ecfc, avg_reads=5)
top.s1etv2.ecfc <- topEvents(f.s1etv2.ecfc, fdr = 0.05, deltaPSI = 0.1)
```

```{r ips.s1s2 opt decluttering}
#remove raw Maser objects
rm(ips.s1s2, ips.s1etv2, s1etv2.ecfc)
rm(ips.s1s2.path, ips.s1etv2.path, s1etv2.ecfc.path)

#view filtered Maser object
#head(summery(f.ips.s1s2))

#interactive table of objects by event type
#maser::display(f.ips.s1s2, "SE")
```

```{r topevents n_distinct}
top.ips.s1s2.table <- summary(top.ips.s1s2) %>% as.data.frame()

top.ips.s1etv2.table <- summary(top.ips.s1etv2) %>% as.data.frame()
top.s1etv2.ecfc.table <- summary(top.s1etv2.ecfc) %>% as.data.frame()

n_distinct(top.ips.s1s2.table$GeneID)
n_distinct(top.ips.s1etv2.table$GeneID)
n_distinct(top.s1etv2.ecfc.table$GeneID)
```

```{r}
#PREPROCESSING FOR MAPPIN SCRIPT 10/22/22
top.ips.s1s2 <- ips.s1s2
SE.PSI.Pval <- summary(top.ips.s1s2, type = "SE")[] %>% as.data.frame() %>% mutate("EventType" = rep("SE"))
MXE.PSI.Pval <- summary(top.ips.s1s2, type = "MXE")[] %>% as.data.frame() %>% mutate("EventType" = rep("MXE"))
RI.PSI.Pval <- summary(top.ips.s1s2, type = "RI")[] %>% as.data.frame() %>% mutate("EventType" = rep("RI"))
A5SS.PSI.Pval <- summary(top.ips.s1s2, type = "A5SS")[] %>% as.data.frame() %>% mutate("EventType" = rep("A5SS"))
A3SS.PSI.Pval <- summary(top.ips.s1s2, type = "A3SS")[] %>% as.data.frame() %>% mutate("EventType" = rep("A3SS"))
```


```{r}
SE <- summary(top.ips.s1s2, type = "SE")[] %>% as.data.frame() %>% mutate("EventType" = rep("SE")) %>% head(10)
MXE <- summary(top.ips.s1s2, type = "MXE")[] %>% as.data.frame() %>% mutate("EventType" = rep("MXE")) %>% head(10)
RI <- summary(top.ips.s1s2, type = "RI")[] %>% as.data.frame() %>% mutate("EventType" = rep("RI")) %>% head(10)
A5SS <- summary(top.ips.s1s2, type = "A5SS")[] %>% as.data.frame() %>% mutate("EventType" = rep("A5SS")) %>% head(10)
A3SS <- summary(top.ips.s1s2, type = "A3SS")[] %>% as.data.frame() %>% mutate("EventType" = rep("A3SS")) %>% head(10)


SE %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/seToy.csv")
MXE %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/mxeToy.csv")
RI %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/riToy.csv")
A5SS %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/a5ssToy.csv")
A3SS %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/a3ssToy.csv")

```

```{r ips.s1s2 masterlist creation}
#filtering all events in the Maser object by AS event type
#adding column that specifies the AS event type
SE.PSI.Pval <- summary(top.ips.s1s2, type = "SE")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("SE"))
MXE.PSI.Pval <- summary(top.ips.s1s2, type = "MXE")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("MXE"))
RI.PSI.Pval <- summary(top.ips.s1s2, type = "RI")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("RI"))
A5SS.PSI.Pval <- summary(top.ips.s1s2, type = "A5SS")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("A5SS"))
A3SS.PSI.Pval <- summary(top.ips.s1s2, type = "A3SS")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("A3SS"))

#combining into master data set
master.ips.s1s2 <- rbind(SE.PSI.Pval, MXE.PSI.Pval, RI.PSI.Pval, A5SS.PSI.Pval, A3SS.PSI.Pval)

master.ips.s1s2 <- master.ips.s1s2 %>% filter(FDR <= 0.05) %>% filter(PValue <= 0.08)

rm(SE.PSI.Pval, MXE.PSI.Pval, RI.PSI.Pval, A5SS.PSI.Pval, A3SS.PSI.Pval)
```

```{r ips.s1etv2 masterlist creation}
#filtering all events in the Maser object by AS event type
#adding column that specifies the AS event type
SE.PSI.Pval <- summary(top.ips.s1etv2, type = "SE")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("SE"))
MXE.PSI.Pval <- summary(top.ips.s1etv2, type = "MXE")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("MXE"))
RI.PSI.Pval <- summary(top.ips.s1etv2, type = "RI")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("RI"))
A5SS.PSI.Pval <- summary(top.ips.s1etv2, type = "A5SS")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("A5SS"))
A3SS.PSI.Pval <- summary(top.ips.s1etv2, type = "A3SS")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("A3SS"))

#combining into master data set
master.ips.s1etv2 <- rbind(SE.PSI.Pval, MXE.PSI.Pval, RI.PSI.Pval, A5SS.PSI.Pval, A3SS.PSI.Pval)

master.ips.s1etv2 <- master.ips.s1etv2 %>% filter(FDR <= 0.05) %>% filter(PValue <= 0.08)

rm(SE.PSI.Pval, MXE.PSI.Pval, RI.PSI.Pval, A5SS.PSI.Pval, A3SS.PSI.Pval)
```

```{r s1etv2.ecfc masterlist creation}
#filtering all events in the Maser object by AS event type
#adding column that specifies the AS event type
SE.PSI.Pval <- summary(top.s1etv2.ecfc, type = "SE")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("SE"))
MXE.PSI.Pval <- summary(top.s1etv2.ecfc, type = "MXE")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("MXE"))
RI.PSI.Pval <- summary(top.s1etv2.ecfc, type = "RI")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("RI"))
A5SS.PSI.Pval <- summary(top.s1etv2.ecfc, type = "A5SS")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("A5SS"))
A3SS.PSI.Pval <- summary(top.s1etv2.ecfc, type = "A3SS")[,1:8] %>% as.data.frame() %>% mutate("EventType" = rep("A3SS"))

#combining into master data set
master.s1etv2.ecfc <- rbind(SE.PSI.Pval, MXE.PSI.Pval, RI.PSI.Pval, A5SS.PSI.Pval, A3SS.PSI.Pval)

master.s1etv2.ecfc <- master.s1etv2.ecfc %>% filter(FDR <= 0.05) %>% filter(PValue <= 0.08)

rm(SE.PSI.Pval, MXE.PSI.Pval, RI.PSI.Pval, A5SS.PSI.Pval, A3SS.PSI.Pval)
```

```{r}
gtf_path <- "/Volumes/sheynkman/projects/reference-files/gencode.v38.annotation.gtf"
ens_gtf <- rtracklayer::import.gff(gtf_path)
```

```{r}
master.ips.s1s2 %>% filter(geneSymbol == "FOXP1")

foxp1_events <- geneEvents(f.ips.s1s2, geneS = "FOXP1", fdr = 0.1, deltaPSI = 0.1)

plotTranscripts(foxp1_events, event_id = 54673, type = "SE", gtf = ens_gtf, zoom = T, show_PSI = TRUE)
plotTranscripts(foxp1_events, event_id = 54673, type = "SE", gtf = ens_gtf, zoom = F, show_PSI = TRUE)

plotTranscripts(foxp1_events, event_id = 54677, type = "SE", gtf = ens_gtf, zoom = TRUE, show_PSI = TRUE) 
plotTranscripts(foxp1_events, event_id = 54677, type = "SE", gtf = ens_gtf, zoom = F, show_PSI = TRUE) 

plotTranscripts(foxp1_events, event_id = 6043, type = "MXE", gtf = ens_gtf, zoom = TRUE, show_PSI = TRUE) 
plotTranscripts(foxp1_events,  event_id = 6043, type = "MXE", gtf = ens_gtf, zoom = F, show_PSI = TRUE) 

plotTranscripts(foxp1_events, event_id = 5341, type = "A5SS", gtf = ens_gtf, zoom = TRUE, show_PSI = TRUE) 
plotTranscripts(foxp1_events, event_id = 5341, type = "A5SS", gtf = ens_gtf, zoom = F, show_PSI = TRUE) 

plotTranscripts(foxp1_events, event_id = 7984, type = "A3SS", gtf = ens_gtf, zoom = TRUE, show_PSI = TRUE)
plotTranscripts(foxp1_events, event_id = 7984, type = "A3SS", gtf = ens_gtf, zoom = F, show_PSI = TRUE) 
```

```{r}
master.ips.s1etv2 %>% filter(geneSymbol == "FOXP1")

foxp1_events <- geneEvents(f.ips.s1etv2, geneS = "FOXP1", fdr = 0.1, deltaPSI = 0.1)

plotTranscripts(foxp1_events, event_id = 56617, type = "SE", gtf = ens_gtf, zoom = T, show_PSI = TRUE)
plotTranscripts(foxp1_events, event_id = 56617, type = "SE", gtf = ens_gtf, zoom = F, show_PSI = TRUE)

plotTranscripts(foxp1_events, event_id = 56620, type = "SE", gtf = ens_gtf, zoom = TRUE, show_PSI = TRUE) 
plotTranscripts(foxp1_events, event_id = 56620, type = "SE", gtf = ens_gtf, zoom = F, show_PSI = TRUE) 

plotTranscripts(foxp1_events, event_id = 6516, type = "MXE", gtf = ens_gtf, zoom = TRUE, show_PSI = TRUE) 
plotTranscripts(foxp1_events,  event_id = 6516, type = "MXE", gtf = ens_gtf, zoom = F, show_PSI = TRUE)

plotTranscripts(foxp1_events, event_id = 5373, type = "A5SS", gtf = ens_gtf, zoom = TRUE, show_PSI = TRUE) 
plotTranscripts(foxp1_events, event_id = 5373, type = "A5SS", gtf = ens_gtf, zoom = F, show_PSI = TRUE) 

plotTranscripts(foxp1_events, event_id = 7999, type = "A3SS", gtf = ens_gtf, zoom = TRUE, show_PSI = TRUE)
plotTranscripts(foxp1_events, event_id = 7999, type = "A3SS", gtf = ens_gtf, zoom = F, show_PSI = TRUE) 
```

```{r}
master.s1etv2.ecfc %>% filter(geneSymbol == "FOXP1")

foxp1_events <- geneEvents(f.s1etv2.ecfc, geneS = "FOXP1", fdr = 0.1, deltaPSI = 0.1)

plotTranscripts(foxp1_events, event_id = 56620, type = "SE", gtf = ens_gtf, zoom = TRUE, show_PSI = TRUE) 
```

```{r}
sig.ips.s1s2 <- master.ips.s1s2 %>% filter(abs(IncLevelDifference) >= 0.5)
sig.ips.s1etv2 <- master.ips.s1etv2 %>% filter(abs(IncLevelDifference) >= 0.5)
sig.s1etv2.ecfc <- master.s1etv2.ecfc %>% filter(abs(IncLevelDifference) >= 0.5)
```

```{r eval = FALSE}
typecounts.ips.s1s2.5 <- master.ips.s1s2 %>% group_by(geneSymbol, GeneID, EventType) %>% 
  summarize("NumEventType" = n()) %>% 
  pivot_wider(names_from = EventType, values_from = NumEventType) %>% 
  replace(is.na(.), 0) %>% 
  filter(abs(IncLevelDifference) >= 0.5) %>% 
  mutate(
    "TotalNumEvents" = sum(RI, SE, A3SS, A5SS, MXE),
    "dPSI.Threshold" = 0.5
  )
typecounts.ips.s1etv2.5 <- master.ips.s1etv2 %>% group_by(geneSymbol, GeneID, EventType) %>% 
  summarize("NumEventType" = n()) %>% 
  pivot_wider(names_from = EventType, values_from = NumEventType) %>% 
  replace(is.na(.), 0) %>% 
  filter(abs(IncLevelDifference) >= 0.5) %>% 
  mutate(
    "TotalNumEvents" = sum(RI, SE, A3SS, A5SS, MXE),
    "dPSI.Threshold" = 0.5
  )
typecounts.s1etv2.ecfc.5 <- master.s1etv2.ecfc %>% group_by(geneSymbol, GeneID, EventType) %>% 
  summarize("NumEventType" = n()) %>% 
  pivot_wider(names_from = EventType, values_from = NumEventType) %>% 
  replace(is.na(.), 0) %>% 
  filter(abs(IncLevelDifference) >= 0.5) %>% 
  mutate(
    "TotalNumEvents" = sum(RI, SE, A3SS, A5SS, MXE),
    "dPSI.Threshold" = 0.5
  )
```

```{r eval = FALSE}
typecounts.ips.s1s2.1 <- master.ips.s1s2 %>% group_by(geneSymbol, GeneID, EventType) %>% 
  summarize("NumEventType" = n()) %>% 
  pivot_wider(names_from = EventType, values_from = NumEventType) %>% 
  replace(is.na(.), 0) %>% 
  mutate(
    "TotalNumEvents" = sum(RI, SE, A3SS, A5SS, MXE),
    "dPSI.Threshold" = 0.1
  ) %>% 
  filter(abs(IncLevelDifference) >= 0.1)
typecounts.ips.s1etv2.1 <- master.ips.s1etv2 %>% group_by(geneSymbol, GeneID, EventType) %>% 
  summarize("NumEventType" = n()) %>% 
  pivot_wider(names_from = EventType, values_from = NumEventType) %>% 
  replace(is.na(.), 0) %>% 
  mutate(
    "TotalNumEvents" = sum(RI, SE, A3SS, A5SS, MXE),
    "dPSI.Threshold" = 0.1
  )%>% 
  filter(abs(IncLevelDifference) >= 0.1)

typecounts.s1etv2.ecfc.1 <- master.s1etv2.ecfc %>% group_by(geneSymbol, GeneID, EventType) %>% 
  summarize("NumEventType" = n()) %>% 
  pivot_wider(names_from = EventType, values_from = NumEventType) %>% 
  replace(is.na(.), 0) %>% 
  mutate(
    "TotalNumEvents" = sum(RI, SE, A3SS, A5SS, MXE),
    "dPSI.Threshold" = 0.1
  )%>% 
  filter(abs(IncLevelDifference) >= 0.1) 
```

```{r}
total.ips.s1s2 <- 1896
total.ips.s1etv2 <- 1919
total.s1etv2.ecfc <- 480

prop.ips.s1s2 <- master.ips.s1s2 %>% group_by(EventType) %>% summarize("prop" = (n()/total.ips.s1s2)*100)

prop.ips.s1etv2 <- master.ips.s1etv2 %>% group_by(EventType) %>% summarize("prop" = (n()/total.ips.s1etv2)*100)

prop.s1etv2.ecfc <- master.s1etv2.ecfc %>% group_by(EventType) %>% summarize("prop" = (n()/total.s1etv2.ecfc)*100)
```

```{r}
shared.ENSGs <- intersect(master.ips.s1s2$GeneID,master.ips.s1etv2$GeneID) %>% as_vector() 

shared.ips.s1s2 <- master.ips.s1s2 %>% filter(GeneID %in% shared.ENSGs) %>% mutate("Source" = "ips.s1s2")
shared.ips.s1etv2 <- master.ips.s1etv2 %>% filter(GeneID %in% shared.ENSGs) %>% mutate("Source" = "ips.s1etv2")

shared.ips.s1s2etv2 <- rbind(shared.ips.s1s2, shared.ips.s1etv2)
```

```{r}
shared.ips.s1s2etv2 %>% write.csv(file = "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/000_data/shared.ENSGs.events.s1s2.s1etv2.csv")
```

```{r}
mouse.to.human <- read.csv("/Volumes/sheynkman/projects/EC_stem_cell_differentiation/000_data/mouse-human-orthologs-EC-differentiation.csv") %>% as.data.frame()

humangenename <- mouse.to.human$Human.gene.name %>% unique()

m1 <- master.ips.s1s2 %>% filter(geneSymbol %in% humangenename) %>% mutate("dataset" = "ips.s1s2")
m2 <- master.ips.s1etv2 %>% filter(geneSymbol %in% humangenename) %>% mutate("dataset" = "ips.s1etv2")
m3 <- master.s1etv2.ecfc %>% filter(geneSymbol %in% humangenename) %>% mutate("dataset" = "s1etv2.ecfc")

genes.in.mouselist <- rbind(m1, m2, m3)
```

```{r eval = FALSE}
write.csv(x = master.ips.s1s2, file = "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/001_ips-s1s2/003_ips-s1s2_output/003_ips-s1s2_maser-out/master.ips.s1s2.csv")
write.csv(x = master.ips.s1etv2, file = "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/002_ips-s1etv2/003_ips-s1etv2_output/003_ips-s1etv2_maser-out/master.ips.s1etv2.csv")
write.csv(x = master.s1etv2.ecfc, file = "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/003_s1etv2-ecfc/003_s1etv2-ecfc_output/003_s1etv2-ecfc_maser-out/master.s1etv2.ecfc.csv")

write.csv(x = sig.ips.s1s2, file = "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/001_ips-s1s2/003_ips-s1s2_output/003_ips-s1s2_maser-out/sig.ips.s1s2.csv")
write.csv(x = sig.ips.s1etv2, file = "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/002_ips-s1etv2/003_ips-s1etv2_output/003_ips-s1etv2_maser-out/sig.ips.s1etv2.csv")
write.csv(x = sig.s1etv2.ecfc, file = "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/003_s1etv2-ecfc/003_s1etv2-ecfc_output/003_s1etv2-ecfc_maser-out/sig.s1etv2.ecfc.csv")

write.csv(x = typecounts.ips.s1s2.5, file = "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/001_ips-s1s2/003_ips-s1s2_output/003_ips-s1s2_maser-out/typecounts.ips.s1s2.5.csv")
write.csv(x = typecounts.ips.s1etv2.5, file = "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/002_ips-s1etv2/003_ips-s1etv2_output/003_ips-s1etv2_maser-out/typecounts.ips.s1etv2.5.csv")
write.csv(x = typecounts.s1etv2.ecfc.5, file = "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/003_s1etv2-ecfc/003_s1etv2-ecfc_output/003_s1etv2-ecfc_maser-out/typecounts.s1etv2.ecfc.5.csv")

write.csv(x = typecounts.ips.s1s2.1, file = "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/001_ips-s1s2/003_ips-s1s2_output/003_ips-s1s2_maser-out/typecounts.ips.s1s2.1.csv")
write.csv(x = typecounts.ips.s1etv2.1, file = "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/002_ips-s1etv2/003_ips-s1etv2_output/003_ips-s1etv2_maser-out/typecounts.ips.s1etv2.1.csv")
write.csv(x = typecounts.s1etv2.ecfc.1, file = "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/003_s1etv2-ecfc/003_s1etv2-ecfc_output/003_s1etv2-ecfc_maser-out/typecounts.s1etv2.ecfc.1.csv")

write.csv(x = genes.in.mouselist, file = "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/000_data/genes.in.mouselist.csv")
```

```{r}
master.ips.s1etv2 %>% filter(geneSymbol == "BAZ2B")

baz2b_events <- geneEvents(f.ips.s1etv2, geneS = "BAZ2B", fdr = 0.1, deltaPSI = 0.1)

#plotTranscripts(baz2b_events, event_id = 72456, type = "SE", gtf = ens_gtf, zoom = T, show_PSI = TRUE)

baz2b_mapped <- mapTranscriptsToEvents(baz2b_events, ens_gtf)

baz2b_annot <- mapProteinFeaturesToEvents(baz2b_mapped, c("Domain_and_Sites", "Topology"), by = "category")

plotUniprotKBFeatures(baz2b_mapped, "SE", event_id = 72456, gtf = ens_gtf, features = c("coiled", "region"), show_transcripts = TRUE)
```

```{r}
foxp1_events <- geneEvents(f.ips.s1etv2, "FOXP1")
foxp1_events

foxp1_mapped <- mapTranscriptsToEvents(foxp1_events, ens_gtf)

head(annotation(foxp1_mapped, "SE"))

foxp1_annot <- mapProteinFeaturesToEvents(foxp1_mapped, c("Domain_and_Sites", "Topology"), by="category")

head(annotation(foxp1_annot, "SE"))

plotUniprotKBFeatures(foxp1_mapped, "SE", event_id = 56620, gtf = ens_gtf, 
   features = c("domain", "chain"), show_transcripts = TRUE)
```

```{r}
master.ips.s1s2 %>% filter(geneSymbol == "BAZ2B")

baz2b_events <- geneEvents(f.ips.s1s2, geneS = "BAZ2B", fdr = 0.1, deltaPSI = 0.1)

#plotTranscripts(baz2b_events, event_id = 72456, type = "SE", gtf = ens_gtf, zoom = T, show_PSI = TRUE)

baz2b_mapped <- mapTranscriptsToEvents(baz2b_events, ens_gtf)

baz2b_annot <- mapProteinFeaturesToEvents(baz2b_mapped, c("Domain_and_Sites", "Topology"), by = "category")

head(annotation(baz2b_annot, "SE"))

plotUniprotKBFeatures(baz2b_mapped, "SE", event_id = 69953, gtf = ens_gtf, features = c("coiled", "region"), show_transcripts = TRUE)
```

```{r}
celf1_events <- geneEvents(f.ips.s1etv2, "CELF1")
celf1_events
summary(celf1_events)
celf1_mapped <- mapTranscriptsToEvents(celf1_events, ens_gtf)

head(annotation(celf1_mapped, "A3SS"))

foxp1_annot <- mapProteinFeaturesToEvents(foxp1_mapped, c("Domain_and_Sites", "Topology"), by="category")

head(annotation(foxp1_annot, "SE"))

plotUniprotKBFeatures(foxp1_mapped, "SE", event_id = 56620, gtf = ens_gtf, 
   features = c("domain", "chain"), show_transcripts = TRUE)
```


```{r comparative volcano plots}
se.v <- volcano(f.ips.s1s2, fdr = 0.05, deltaPSI = 0.1, type = "SE") + 
  ggtitle("Overview of significant Skipped Exon type events (hiPSCs vs S1S2)", subtitle = "FDR <= 0.05, deltaPSI >= 0.1")
mxe.v <- volcano(f.ips.s1s2, fdr = 0.05, deltaPSI = 0.1, type = "MXE") + 
  ggtitle("Overview of significant Mutually Exclusive type events (hiPSCs vs S1S2)", subtitle = "FDR <= 0.05, deltaPSI >= 0.1")
ri.v <- volcano(f.ips.s1s2, fdr = 0.05, deltaPSI = 0.1, type = "RI") + 
  ggtitle("Overview of significant Retained Intron type events (hiPSCs vs S1S2)", subtitle = "FDR <= 0.05, deltaPSI >= 0.1")
a3ss.v <- volcano(f.ips.s1s2, fdr = 0.05, deltaPSI = 0.1, type = "A3SS") + 
  ggtitle("Overview of significant Alternative 3' Splice Site type events (hiPSCs vs S1S2)", subtitle = "FDR <= 0.05, deltaPSI >= 0.1")
a5ss.v <- volcano(f.ips.s1s2, fdr = 0.05, deltaPSI = 0.1, type = "A5SS") + 
  ggtitle("Overview of significant Alternative 5' Splice Site type events (hiPSCs vs S1S2)", subtitle = "FDR <= 0.05, deltaPSI >= 0.1")

grid.arrange(se.v, mxe.v, ri.v, a3ss.v, a5ss.v, ncol = 2)
```


```{r comparative volcano plots}
se.v <- volcano(f.ips.s1etv2, fdr = 0.05, deltaPSI = 0.1, type = "SE") + 
  ggtitle("Overview of significant Skipped Exon type events (hiPSCs vs S1modETV2)", subtitle = "FDR <= 0.05, deltaPSI >= 0.1")
mxe.v <- volcano(f.ips.s1etv2, fdr = 0.05, deltaPSI = 0.1, type = "MXE") + 
  ggtitle("Overview of significant Mutually Exclusive type events (hiPSCs vs S1modETV2)", subtitle = "FDR <= 0.05, deltaPSI >= 0.1")
ri.v <- volcano(f.ips.s1etv2, fdr = 0.05, deltaPSI = 0.1, type = "RI") + 
  ggtitle("Overview of significant Retained Intron type events (hiPSCs vs S1modETV2)", subtitle = "FDR <= 0.05, deltaPSI >= 0.1")
a3ss.v <- volcano(f.ips.s1etv2, fdr = 0.05, deltaPSI = 0.1, type = "A3SS") + 
  ggtitle("Overview of significant Alternative 3' Splice Site type events (hiPSCs vs S1modETV2)", subtitle = "FDR <= 0.05, deltaPSI >= 0.1")
a5ss.v <- volcano(f.ips.s1etv2, fdr = 0.05, deltaPSI = 0.1, type = "A5SS") + 
  ggtitle("Overview of significant Alternative 5' Splice Site type events (hiPSCs vs S1modETV2)", subtitle = "FDR <= 0.05, deltaPSI >= 0.1")

grid.arrange(se.v, mxe.v, ri.v, a3ss.v, a5ss.v, ncol = 2)
```

```{r}
head(summary(f.ips.s1s2))
```

