---
output: html_document
editor_options: 
  chunk_output_type: console
---
...
title: "rMATS analysis"
author: "Shay Iyer"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
...

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(maser)
library(rtracklayer)
library(gridExtra) 
```

```{r maser path definition}
ips.s1s2.path <- "/Volumes/sheynkman/projects/EC_stem_cell_differentiation/001_ips-s1s2/003_ips-s1s2_output/002_ips-s1s2_rmats-out"
```

```{r maser objects definition}
#using JCEC for statistical power
ips.s1s2 <- maser(path = ips.s1s2.path, cond_labels = c("iPS", "S1S2-iEC"), ftype = "JCEC")
```

# PreProcessing for mapping script, 10/22/22

```{r}
SE <- summary(ips.s1s2, type = "SE")[] %>% as.data.frame() %>% mutate("EventType" = rep("SE")) %>% filter(Chr == "chr19") %>% filter(FDR < 0.05)
MXE <- summary(ips.s1s2, type = "MXE")[] %>% as.data.frame() %>% mutate("EventType" = rep("MXE")) %>% filter(Chr == "chr19") %>% filter(FDR < 0.05)
A5SS <- summary(ips.s1s2, type = "A5SS")[] %>% as.data.frame() %>% mutate("EventType" = rep("A5SS"))%>% filter(Chr == "chr19") %>% filter(FDR < 0.05)
A3SS <- summary(ips.s1s2, type = "A3SS")[] %>% as.data.frame() %>% mutate("EventType" = rep("A3SS")) %>% filter(Chr == "chr19") %>% filter(FDR < 0.05)
```

```{r}
rm(ips.s1s2, ips.s1s2.path)
```

```{r}
SE %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/seToy.csv")
MXE %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/mxeToy.csv")
A5SS %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/a5ssToy.csv")
A3SS %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/a3ssToy.csv")
```

```{r not using retained intron for 10/23/22}
RI <- summary(ips.s1s2, type = "RI")[] %>% as.data.frame() %>% mutate("EventType" = rep("RI")) 
RI %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/riToy.csv")
```


```{r}
EC.transcript.annotation.path <- "/Volumes/sheynkman/projects/shay_thesis/data/EC-LR/long-read-EC-data/03_chr19_gtfs/chr19_EC.gtf"

EC.ta.gtf <- rtracklayer::import(EC.transcript.annotation.path)

EC.ta <- EC.ta.gtf %>% as.data.frame()

rm(EC.trancript.annotation.path)

rm(EC.ta.gtf)

EC.ta = EC.ta %>% as.data.frame() %>% select(seqnames, start, end, width, strand, type, gene_id, gene_name, exon_id, exon_number, transcript_id, transcript_name)

EC.ta %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/EC-transcript-annotation.csv")
```

```{r}
WTC11.1.transcript.annotation.path <- "/Volumes/sheynkman/projects/shay_thesis/data/EC-LR/long-read-EC-data/03_chr19_gtfs/chr19_WTC11-1.gtf"

WTC11.1.ta.gtf <- rtracklayer::import(WTC11.1.transcript.annotation.path)

WTC11.1.ta <- WTC11.1.ta.gtf %>% as.data.frame()

rm(WTC11.1.trancript.annotation.path)

rm(WTC11.1.ta.gtf)

WTC11.1.ta = WTC11.1.ta %>% as.data.frame() %>% select(seqnames, start, end, width, strand, type, gene_id, gene_name, exon_id, exon_number, transcript_id, transcript_name)

WTC11.1.ta %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/WTC11-1-transcript-annotation.csv")
```

```{r}
WTC11.2.transcript.annotation.path <- "/Volumes/sheynkman/projects/shay_thesis/data/EC-LR/long-read-EC-data/03_chr19_gtfs/chr19_WTC11-2.gtf"

WTC11.2.ta.gtf <- rtracklayer::import(WTC11.2.transcript.annotation.path)

WTC11.2.ta <- WTC11.2.ta.gtf %>% as.data.frame()

rm(WTC11.2.trancript.annotation.path)

rm(WTC11.2.ta.gtf)

WTC11.2.ta = WTC11.2.ta %>% as.data.frame() %>% select(seqnames, start, end, width, strand, type, gene_id, gene_name, exon_id, exon_number, transcript_id, transcript_name)

WTC11.2.ta %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/WTC11-2-transcript-annotation.csv")
```

```{r}
WTC11.3.transcript.annotation.path <- "/Volumes/sheynkman/projects/shay_thesis/data/EC-LR/long-read-EC-data/03_chr19_gtfs/chr19_WTC11-3.gtf"

WTC11.3.ta.gtf <- rtracklayer::import(WTC11.3.transcript.annotation.path)

WTC11.3.ta <- WTC11.3.ta.gtf %>% as.data.frame()

rm(WTC11.3.trancript.annotation.path)

rm(WTC11.3.ta.gtf)

WTC11.3.ta = WTC11.3.ta %>% as.data.frame() %>% select(seqnames, start, end, width, strand, type, gene_id, gene_name, exon_id, exon_number, transcript_id, transcript_name)

WTC11.3.ta %>% write.csv(file = "/Volumes/sheynkman/projects/shay_thesis/data/chr19-lr-proc/WTC11-3-transcript-annotation.csv")
```
